File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.cpp
#include <iostream>
#include <vector>
#include <stack>
#include <string>
using namespace std;
stack <char> v;
int solution(string s){
    for(char c : s){
        if(!v.empty()){
            if(v.top() == c) v.pop();
             else v.push(c);
        }
        else v.push(c);
    }
    if(v.empty()) return 1;
    return 0;
}

File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    stack = []
    for now in s:
        if len(stack) and stack[-1] == now:
            stack.pop()
        else:
            stack.append(now)
    
    if len(stack) == 0:
        answer = 1
    else:
        answer = 0
    

    return answer